object.go

```diff
@@ -275,27 +275,36 @@ func (d *Decoder) decClassDef() (interface{}, error) {
 	return classInfo{javaName: clsName, fieldNameList: fieldList}, nil
 }
 
-func findField(name string, typ reflect.Type) (int, error) {
+func findField(name string, typ reflect.Type) ([]int, error) {
 	for i := 0; i < typ.NumField(); i++ {
 		// matching tag first, then lowerCamelCase, SameCase, lowerCase
 
-		if val, has := typ.Field(i).Tag.Lookup(tagIdentifier); has && strings.Compare(val, name) == 0 {
-			return i, nil
+		typField := typ.Field(i)
+
+		if val, has := typField.Tag.Lookup(tagIdentifier); has && strings.Compare(val, name) == 0 {
+			return []int{i}, nil
 		}
 
-		fieldName := typ.Field(i).Name
+		fieldName := typField.Name
 		switch {
 		case strings.Compare(lowerCamelCase(fieldName), name) == 0:
-			return i, nil
+			return []int{i}, nil
 		case strings.Compare(fieldName, name) == 0:
-			return i, nil
+			return []int{i}, nil
 		case strings.Compare(strings.ToLower(fieldName), name) == 0:
-			return i, nil
+			return []int{i}, nil
 		}
 
+		if typField.Anonymous && typField.Type.Kind() == reflect.Struct {
+			next, _ := findField(name, typField.Type)
+			if len(next) > 0 {
+				pos := []int{i}
+				return append(pos, next...), nil
+			}
+		}
 	}
 
-	return 0, perrors.Errorf("failed to find field %s", name)
+	return []int{}, perrors.Errorf("failed to find field %s", name)
```

Comment:
Is it necessary to return error ?

for example, the server api upgrade and add a new field, the client code would upgrade later. between this time, the client will don't work

another case, if json bytes has more fields, json.unmarshal will ignore the unused fields and don't report any error.

Reply:
not like json or pb, hessian can't ignore fields, return error is necessary.

Reply:
yes， gohessian can ignore fields, but not the default behavior,  to do that u should check decoder.isSkip